docs/berkeley-upgrade/migrating-archive-database-to-berkeley.mdx

```diff
@@ -13,182 +13,313 @@ keywords:
 
-### Example: Downloading schema sources from GitHub
+`-bs | --blocks-batch-size`: *number of precomputed blocks to be fetch at once from Gcloud. Bigger number like 1000 can help speed up migration process*
 
-   ```sh
-   wget https://raw.githubusercontent.com/MinaProtocol/mina/berkeley/src/app/archive/zkapp_tables.sql
+`-n | --network`: *network name when determining precomputed blocks. NOTICE: there is an assumption that precomputed blocks are named with format: `{network}-{height}-{state_hash}.json`*
+  
+:info:
 
-   wget https://raw.githubusercontent.com/MinaProtocol/mina/berkeley/src/app/archive/create_schema.sql
+migration-checkpoint.json - is a last replayer checkpoint from initial run
 
-   ```
+#### Stage 3: Remainder migration
 
-### Migration process
+```
+./scripts/archive/migration/berkeley_migration.sh \
+   final \
+   --genesis-ledger ledger.json \
+   --source-db postgres://postgres:postgres@localhost:5432/source \
+   --target-db postgres://postgres:postgres@localhost:5432/migrated \
+   --blocks-bucket mina_network_block_data \
+   --blocks-batch-size 50 \
+   --network mainnet \
+   --replayer-checkpoint migration-checkpoint.json \
+   -fc fork-dump.json \
+   -f fork-state-hash 
+```
 
-As mentioned in [Planning the Archive Migration](/berkeley-upgrade/planning-archive-migration), the mainnet migration can take up to a couple of days. 
-Therefore, it is recommended to split migration into 3 stages:
+where: 
 
-- **Stage1:** Initial migration
+`-r | --replayer-checkpoint`: *path to replayer checkpoint file*
 
-- **Stage2:** Incremental migration
+`-g | --genesis-ledger`: *path to genesis ledger file*
 
-- **Stage3:** Remainder migration
+`-f | --fork-state-hash`: *fork state hash*
 
-Each stage has two migration phases:
+`-s | --source-db`: *connection string to database to be migrated*
 
-- **Phase 1:** Copying data from mainnet database and precomputed blocks using the **berkeley_migration** app.
+`-t | --target-db`: *connection string to database which will hold migrated data*
 
-- **Phase 2:** Populating new berkeley tables using the **replayer app in migration mode**
+`-b | --blocks-bucket`: *name of precomputed blocks bucket. NOTICE: there is an assumption that precomputed blocks are named with format: `{network}-{height}-{state_hash}.json`*
 
-Review these phases and stages before you start the migration:
+`-bs | --blocks-batch-size`: *number of precomputed blocks to be fetch at once from Gcloud. Bigger number like 1000 can help speed up migration process*
+
+`-n | --network`: *network name when determining precomputed blocks. NOTICE: there is an assumption that precomputed blocks are named with format: `{network}-{height}-{state_hash}.json`*
+
+`-fc | --fork-config`: *Fork-state config file is dump file in form of json containing fork block and ledger file. It should be provied by MF or O(1)Labs team after fork block is announced*
```

Comment:
*Fork-genesis-config file is the new genesis config. It would be distributed with the new daemon and would also be published after fork block is announced.